src/socket.cpp

```diff
@@ -79,9 +79,10 @@ void Socket::connect(const char _addr[], const int _port) const
 
 	// TODO: Check for error == EINPROGRESS and don't throw exception in that
 	// case.
+	// CHECK if this is the right way to complete this TODO
 
-	if (res == -1) {
-		throw std::runtime_error(net::method::getErrorMsg());
+	if (res == -1 && errno != EINPROGRESS ) {
```

Comment:
Also, needs `#include <errno.h>` for errno - http://man7.org/linux/man-pages/man3/errno.3.html